# Лабораторная работа №1

## Выполнил Лапатин Данила гр.932001

In [17]:
import pandas as pd

In [16]:
#открываем файл с данными и выводим на экран
df = pd.read_csv("HousePrices.csv", index_col=0)

In [15]:
#типы значений атрибутов
datatypes = df.dtypes

In [14]:
df

price  lotsize  bedrooms  bathrooms  stories driveway recreation   
1     42000.0     5850         3          1        2      yes         no  \
2     38500.0     4000         2          1        1      yes         no   
3     49500.0     3060         3          1        1      yes         no   
4     60500.0     6650         3          1        2      yes        yes   
5     61000.0     6360         2          1        1      yes         no   
..        ...      ...       ...        ...      ...      ...        ...   
542   91500.0     4800         3          2        4      yes        yes   
543   94000.0     6000         3          2        4      yes         no   
544  103000.0     6000         3          2        4      yes        yes   
545  105000.0     6000         3          2        2      yes        yes   
546  105000.0     6000         3          1        2      yes         no   

    fullbase gasheat aircon  garage prefer  
1        yes      no     no       1     no  
2         no      no     no       0     no  
3         no      no     no       0     no  
4         no      no     no       0     no  
5         no      no     no       0     no  
..       ...     ...    ...     ...    ...  
542       no      no    yes       0     no  
543       no      no    yes       0     no  
544       no      no    yes       1     no  
545       no      no    yes       1     no  
546       no      no    yes       1     no  

[546 rows x 12 columns]

# Описание

## Атрибуты:
1. price - Цена продажи дома.
2. lotsize - Площадь участка недвижимости в квадратных футах.
3. bedrooms - Количество спален.
4. bathrooms - Количество полностью оборудованных ванных комнат.
5. stories - Количество этажей без учета подвала.
6. driveway - Есть ли у дома подъездная дорожка
7. recreation - Есть ли в доме комната отдыха
8. fullbase - Есть ли в доме полностью законченный подвал
9. gasheat - Используется ли в доме газ для нагрева горячей воды
10. aircon - Есть ли центральное кондиционирование воздуха
11. garage - Количество мест в гараже.
12. prefer - Расположен ли дом в предпочтительном районе города

### Цены продажи домов, проданных в городе Виндзор, Канада, в июле, августе и сентябре 1987 года.

 Датасет состоит из 546 наблюдений по 12 переменным. Каждая запись - набор характеристик проданного дома.

### Типы значений

In [13]:
datatypes

price         float64
lotsize         int64
bedrooms        int64
bathrooms       int64
stories         int64
driveway       object
recreation     object
fullbase       object
gasheat        object
aircon         object
garage          int64
prefer         object
dtype: object

# Структура данных

In [23]:
#общая информация о датасете
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 546 entries, 1 to 546
Data columns (total 12 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   price       546 non-null    float64
 1   lotsize     546 non-null    int64  
 2   bedrooms    546 non-null    int64  
 3   bathrooms   546 non-null    int64  
 4   stories     546 non-null    int64  
 5   driveway    546 non-null    object 
 6   recreation  546 non-null    object 
 7   fullbase    546 non-null    object 
 8   gasheat     546 non-null    object 
 9   aircon      546 non-null    object 
 10  garage      546 non-null    int64  
 11  prefer      546 non-null    object 
dtypes: float64(1), int64(5), object(6)
memory usage: 55.5+ KB


In [22]:
#описательная статистика числовых данных
df.describe()

price       lotsize    bedrooms   bathrooms     stories   
count     546.000000    546.000000  546.000000  546.000000  546.000000  \
mean    68121.597070   5150.265568    2.965201    1.285714    1.807692   
std     26702.670926   2168.158725    0.737388    0.502158    0.868203   
min     25000.000000   1650.000000    1.000000    1.000000    1.000000   
25%     49125.000000   3600.000000    2.000000    1.000000    1.000000   
50%     62000.000000   4600.000000    3.000000    1.000000    2.000000   
75%     82000.000000   6360.000000    3.000000    2.000000    2.000000   
max    190000.000000  16200.000000    6.000000    4.000000    4.000000   

           garage  
count  546.000000  
mean     0.692308  
std      0.861307  
min      0.000000  
25%      0.000000  
50%      0.000000  
75%      1.000000  
max      3.000000

Mетод DataFrame.describe() для числовых данных включает: count - количество значений, mean - среднее значение, std - среднеквадратическое отклонение, min - минимальное значение, max - максимальное значение, а также нижний (25), 50 и верхний (75) процентили.

In [24]:
#описательная статистика данных типа object
df.describe(include=[object])

driveway recreation fullbase gasheat aircon prefer
count       546        546      546     546    546    546
unique        2          2        2       2      2      2
top         yes         no       no      no     no     no
freq        469        449      355     521    373    418

Mетод DataFrame.describe() для данных типа object включает: count - количество значений, unique - количество уникальных значений, top - наиболее распространённое значение и freq - частота наиболее распространенного значения.

In [25]:
#только первые 5 записей
df.head()

price  lotsize  bedrooms  bathrooms  stories driveway recreation   
1  42000.0     5850         3          1        2      yes         no  \
2  38500.0     4000         2          1        1      yes         no   
3  49500.0     3060         3          1        1      yes         no   
4  60500.0     6650         3          1        2      yes        yes   
5  61000.0     6360         2          1        1      yes         no   

  fullbase gasheat aircon  garage prefer  
1      yes      no     no       1     no  
2       no      no     no       0     no  
3       no      no     no       0     no  
4       no      no     no       0     no  
5       no      no     no       0     no

In [20]:
#только последние 5 записей
df.tail()

price  lotsize  bedrooms  bathrooms  stories driveway recreation   
542   91500.0     4800         3          2        4      yes        yes  \
543   94000.0     6000         3          2        4      yes         no   
544  103000.0     6000         3          2        4      yes        yes   
545  105000.0     6000         3          2        2      yes        yes   
546  105000.0     6000         3          1        2      yes         no   

    fullbase gasheat aircon  garage prefer  
542       no      no    yes       0     no  
543       no      no    yes       0     no  
544       no      no    yes       1     no  
545       no      no    yes       1     no  
546       no      no    yes       1     no

# Работа с пропущенными значениями

In [26]:
#определяем процент NAN в каждом столбце
df.isnull().sum()/df.shape[0]

price         0.0
lotsize       0.0
bedrooms      0.0
bathrooms     0.0
stories       0.0
driveway      0.0
recreation    0.0
fullbase      0.0
gasheat       0.0
aircon        0.0
garage        0.0
prefer        0.0
dtype: float64

Пропущенных значений нет.

Столбцы с процентом пропущенных значений больше 5 обычно просто удаляют, если они не являются значимыми.